### ISSUE TITLE
Sample import menu broken

### ISSUE THREAD

**Issue by gandaja** [democloid/picoTracker#83]
Freezes the unit if I move the cursor north of the [...] symbol or south of the last sample listed. Does not display every sample in samplelib folder.
Edit: does so after upgrading the firmware to v1.1r2

**Comment by democloid** [comment]
@gandaja Can you provide more info about your samplelib folder? Does it have many files? does it work if you reduce the number of files? Do you see similar behaviours navigating up/down the directory tree?

It would be ideal if you can upload your samplelib.

**Comment by gandaja** [comment]
Starting from your last question: the directory tree cycles just fine.
I have three files in my samplelib, named: "bd2", "hh1", "clp1". Only shows the first two. Perhaps the names need to be of the same length?

**Comment by democloid** [comment]
Name length shouldn't be a problem. The fact that only two names appear make me think that there is some invalid reference in the 3rd position where the 3rd file should be and that's why it's locking up going into that position. Not sure what exactly could be the problem, maybe an invalid wav file?

Could you upload those files to check?

**Comment by gandaja** [comment]
[samplelib.zip](https://github.com/democloid/picoTracker/files/13981548/samplelib.zip)

**Comment by maks** [comment]
Thanks for reporting this and uploading the files, I'll take a look at this too as its likely due to the changes I made when optimising the file browser to use less ram to work with directories with more than 25 files.
As @democloid said, file name length shouldn't be an issue, you should be able to use file names of up to 128 characters, but only ASCII, though of course keeping them short makes things easier to see in the picotracker UI.

**Comment by maks** [comment]
@gandaja I just tried your samplelib zip and can reproduce the issue immediately. Its a bit strange since I'm not sure why this is suddenly broken as it doesn't happen for me with same firmware with another sdcard that has large numbers of folders and files that I've been using alot lately. 
I don't have time to look into it right now, but I'll debug this more asap.

**Comment by gandaja** [comment]
Thanks man.
Unsure if relevant, but I'm using a 32gb sdhc card of which I'm attaching a picture

![image](https://github.com/democloid/picoTracker/assets/157061766/bff603ed-cd52-4e05-aa38-4e377a4706e4)

**Comment by maks** [comment]
@gandaja Thanks for adding that. I only had time to very quickly test your files to confirm that I can reproduce the issue you are seeing, but I dont think its to do with sdcard hardware. I think its most likely a bug in handling a small number of files and/or with short names, since when I did my prev changes, I was specifically focused on testing  the opposite: how to handle large numbers of files and nested folders without overflowing available ram, so I likely missed something for the case of only a few files. I also noticed that I havent tested with file on the top level dir vs only having folders, though I dont know if that is part of this bug or not.
